File: src/routing/direct/DirectNetwork.java
package routing.direct;

import node.Node;

import routing.Network;

public class DirectNetwork implements Network {
  public void transfer(Node sender, Node receiver, Block block) {
    receiver.share(block);
  }
  public void connect(Node node1, Node node2){
    node1.link(node2);
    node2.link(node1);
  }
  public void disconnect(Node node1, Node node2){
    node1.unlink(node2);
    node2.unlink(node1);
  }
}

File: src/routing/direct/DirectNetwork.scala
package routing.direct

import node.Node
import routing.Network

class DirectNetwork extends Network {
  override def transfer(sender: Node, receiver: Node, block: Block) {
    receiver.share(block)
  }

  override def connect(node1: Node, node2: Node) {
    node1.link(node2)
    node2.link(node1)
  }

  override def disconnect(node1: Node, node2: Node) {
    node1.unlink(node2)
    node2.unlink(node1)
  }
}